build.gradle.kts

```diff
@@ -426,3 +429,126 @@ tasks.register("allDependencies").configure {
         }
     }
 }
+
+fun getCommittedFilePaths(rootDir: File): List<String> {
+    val filePaths = mutableListOf<String>()
+
+    Git.open(rootDir).use { git ->
+        TreeWalk(git.repository).use { treeWalk ->
+            val headCommit = RevWalk(git.repository).use {
+                val head = git.repository.resolve(Constants.HEAD)
+                it.parseCommit(head)
+            }
+
+            with(treeWalk) {
+                addTree(headCommit.tree)
+                isRecursive = true
+            }
+
+            while (treeWalk.next()) {
+                filePaths += treeWalk.pathString
+            }
+        }
+    }
+
+    return filePaths
+}
+
+fun getCopyrightableFiles(rootDir: File): List<File> {
+    val excludedPaths = listOf(
+        "LICENSE",
+        "NOTICE",
+        "batect",
+        "gradlew",
+        "gradle/",
+        "examples/",
+        "integrations/completions/",
+        "reporter/src/main/resources/",
+        "reporter-web-app/yarn.lock",
+        "resources/META-INF/",
+        "resources/exceptions/",
+        "resources/licenses/",
+        "resources/licenserefs/",
+        "test/assets/",
+        "funTest/assets/"
+    )
+
+    val excludedExtensions = listOf(
+        "css",
+        "graphql",
+        "json",
+        "md",
+        "png",
+        "svg"
+    )
+
+    return getCommittedFilePaths(rootDir).map { filePath ->
+        rootDir.resolve(filePath)
+    }.filter { file ->
+        val isHidden = file.toPath().any { it.toString().startsWith(".") }
+
+        !isHidden && excludedPaths.none { it in file.path } && file.extension !in excludedExtensions
+    }
+}
+
+fun extractCopyrights(file: File): List<String> {
+    val copyrights = mutableListOf<String>()
+
+    val maxLines = 50
+    var lineCounter = 0
+
+    file.useLines { lines ->
+        lines.forEach { line ->
+            if (++lineCounter > maxLines) return@forEach
+            val copyright = line.replaceBefore(" Copyright ", "", "").trim()
+            copyright.takeUnless { it.isEmpty() || it.endsWith("\"") }?.run { copyrights += this }
```

Comment:
A little bit a shorter and a bit more straightforward:
```kotlin
if (copyright.isNotEmpty() && !copyright.endsWith("\"")) copyrights += copyright
```

And maybe `isNotBlank()` instead of `isNotEmpty()`?

Reply:
I've changed the code as suggested but kept `isNotEmpty()` as `isNotBlank()` is not required due to the trimming one line above.